test/extended/operators/olm.go

```diff
@@ -71,3 +74,73 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 		})
 	}
 })
+
+// This context will cover test case: OCP-23440, author: jiazha@redhat.com
+var _ = g.Describe("[Feature:Platform] an end user use OLM", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc           = exutil.NewCLI("olm-23440", exutil.KubeConfigPath())
+		operatorWait = 120 * time.Second
+
+		buildPruningBaseDir = exutil.FixturePath("testdata", "olm")
+		operatorGroup       = filepath.Join(buildPruningBaseDir, "operatorgroup.yaml")
+		etcdSub             = filepath.Join(buildPruningBaseDir, "etcd-subscription.yaml")
+		etcdCluster         = filepath.Join(buildPruningBaseDir, "etcd-cluster.yaml")
+	)
+
+	files := []string{operatorGroup, etcdSub}
+	g.It("can subscribe to the etcd operator", func() {
+		g.By("Cluster-admin user subscribe the operator resource")
+		for _, v := range files {
+			configFile, err := oc.AsAdmin().Run("process").Args("--ignore-unknown-parameters=true", "-f", v, "-p", "NAME=test-operator", fmt.Sprintf("NAMESPACE=%s", oc.Namespace()), "SOURCENAME=community-operators", "SOURCENAMESPACE=openshift-marketplace").OutputToFile("config.json")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = oc.AsAdmin().WithoutNamespace().Run("create").Args("-f", configFile).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+		}
+		err := wait.Poll(10*time.Second, operatorWait, func() (bool, error) {
+			output, err := oc.AsAdmin().Run("get").Args("-n", oc.Namespace(), "csv", "etcdoperator.v0.9.4", "-o=jsonpath={.status.phase}").Output()
+			if err != nil {
+				e2e.Failf("Failed to deploy etcdoperator.v0.9.4, error:%v", err)
+				return false, err
+			}
+			if strings.Contains(output, "Succeeded") {
+				return true, nil
+			}
+			return false, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Switch to common user to create the resources provided by the operator")
+		etcdClusterName := "example-etcd-cluster"
+		configFile, err := oc.Run("process").Args("-f", etcdCluster, "-p", fmt.Sprintf("NAME=%s", etcdClusterName), fmt.Sprintf("NAMESPACE=%s", oc.Namespace())).OutputToFile("config.json")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = oc.Run("create").Args("-f", configFile).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		checkMetrics := map[string]string{
+			"phase":          "Running",
+			"currentVersion": "3.2.13",
+			"size":           "3",
+		}
+
+		for k, v := range checkMetrics {
```

Comment:
You don't really need this loop since `wait.Poll` will churn until timeout, error, or a true return value. You can just check the `phase`, `currentVersion`, and `size` in the closure argument.

Reply:
@njhale Yes, I know. The destination of this loop just to check that three metrics("phase", "currentVersion", "size") and their expected results("Running", "3.2.13", "3"). What do you suggest?

Reply:
@njhale I have removed this loop, thanks!